components/UploadWidget.tsx

```diff
@@ -0,0 +1,277 @@
+import React, { useState } from "react";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { ref, uploadBytesResumable, getDownloadURL } from "firebase/storage";
+import {
+  RecaptchaVerifier,
+  signInWithPhoneNumber,
+  PhoneAuthProvider,
+  signInWithCredential,
+} from "firebase/auth";
+import { storage, auth } from "../firebase-config";
+
+const UploadWidget = () => {
+  const [files, setFiles] = useState([]);
+  const [uploadError, setUploadError] = useState("");
+  const [downloadURLs, setDownloadURLs] = useState([]);
+  const [userName, setUserName] = useState("");
+  const [userPhone, setUserPhone] = useState("");
+  const [otp, setOtp] = useState("");
+  const [showFileInput, setShowFileInput] = useState(true);
+  const [showPhoneInput, setShowPhoneInput] = useState(false);
+  const [showOtpInput, setShowOtpInput] = useState(false);
+  const [showSuccess, setShowSuccess] = useState(false);
+  const [vid, setVid] = useState("");
+
+  const validFileTypes = ["image/jpeg", "image/png", "image/heic"];
+
+  const {
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormData>();
+
+  const handleFileChange = (e) => {
+    if (e.target.files[0]) {
+      const selectedFiles = Array.from(e.target.files)
+        .filter((file) => validFileTypes.includes(file.type))

```

Comment:
@jdeeduk, any idea how to fix this? I get it when I try to deploy.
<img width="593" alt="Screenshot 2024-02-27 at 10 24 37" src="https://github.com/EchoPhotos/Web/assets/8779423/efe526e1-22b7-4013-87d8-8fa966eb34b7">